core/services/gateway/common/aggregation/response_aggregator.go

```diff
@@ -0,0 +1,122 @@
+package aggregation
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	jsonrpc "github.com/smartcontractkit/chainlink-common/pkg/jsonrpc2"
+)
+
+var _ NodeResponseAggregator = (*identicalNodeResponseAggregator)(nil)
+
+type NodeResponseAggregator interface {
+	// CollectAndAggregate appends a node response to existing list of responses if exists
+	// and tries to aggregate them into a single response.
+	CollectAndAggregate(resp *jsonrpc.Response[json.RawMessage], nodeAddress string) (*jsonrpc.Response[json.RawMessage], error)
+}
+
+// identicalNodeResponseAggregator collects node responses and aggregates identical responses.
+// (Usually 2f+1, where f is the number of faulty nodes).
+// NOT thread-safe.
+type identicalNodeResponseAggregator struct {
+	responses map[string]stringSet
+	// nodeToResponse tracks which response key each node address is currently associated with
+	nodeToResponse map[string]string
+	threshold      int
+}
+
+func NewIdenticalNodeResponseAggregator(threshold int) (NodeResponseAggregator, error) {
+	if threshold <= 0 {
+		return nil, fmt.Errorf("threshold must be greater than 0, got %d", threshold)
+	}
+	return &identicalNodeResponseAggregator{
+		responses:      make(map[string]stringSet),
+		nodeToResponse: make(map[string]string),
+		threshold:      threshold,
+	}, nil
+}
+
+// stringSet is a simple set implementation for strings.
+type stringSet map[string]struct{}
+
+func (s stringSet) Add(val string) {
+	s[val] = struct{}{}
+}
+
+func (s stringSet) Contains(val string) bool {
+	_, exists := s[val]
+	return exists
+}
+
+func (s stringSet) Remove(val string) {
+	delete(s, val)
+}
+
+func (s stringSet) Values() []string {
+	values := make([]string, 0, len(s))
+	for k := range s {
+		values = append(values, k)
+	}
+	return values
+}
+
+// TODO: use logic from chainlink-common
```

Comment:
this TODO will be completed once https://github.com/smartcontractkit/chainlink/pull/18509 is merged